extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -66,7 +66,26 @@ CamelRegistryBuildItem registry(
 
             RuntimeValue<Registry> registry = recorder.createRegistry();
 
-            CamelSupport.services(applicationArchives).forEach(si -> {
+            CamelSupport.services(applicationArchives)
+                .filter(si -> {
+                    //
+                    // we need to filter out some services for being registered as some
+                    // extension configure them programmatically and if not properly
+                    // configured, they may cause troubles.
+                    //
+                    // As example, a reactive-executor can be provided by adding a dep on
+                    // camel-reactive-executor-vertx but if that service get detected and
+                    // initialized, it will create an instance of Vert.x during static init
```

Comment:
Its created in doStart lifecycle phase. So I wonder if quarkus-camel is a bit "wrong" if it starts Camel services during build phase? We should only use doInit phase IMHO for that.

Reply:
Yeah, I need to digg into the camel-reactive-executor-vertx a little bit more as the camel context is not supposed to be started at build time but only at runtime init by camel main.

Reply:
ok, found out that I was wrong so I probably had some bad code during the development that leaded to the reactive executor being started in the wrong phase.

I've updated the comment.